File: dataset/ourMethod/CLAUDE_claude-sonnet-4-5-20250929/Task198.java

public class Task198 {
    // Simulated rand7() for testing purposes
    private static int rand7() {
        return (int)(Math.random() * 7) + 1;
    }
    
    public static int rand10() {
        int row, col, idx;
        do {
            row = rand7();
            col = rand7();
            idx = (row - 1) * 7 + col;
        } while (idx > 40);
        return ((idx - 1) % 10) + 1;
    }
    
    public static void main(String[] args) {
        // Test case 1: n = 1
        System.out.print("Test 1: [");
        System.out.print(rand10());
        System.out.println("]");
        
        // Test case 2: n = 2
        System.out.print("Test 2: [");
        for (int i = 0; i < 2; i++) {
            if (i > 0) System.out.print(",");
            System.out.print(rand10());
        }
        System.out.println("]");
        
        // Test case 3: n = 3
        System.out.print("Test 3: [");
        for (int i = 0; i < 3; i++) {
            if (i > 0) System.out.print(",");
            System.out.print(rand10());
        }
        System.out.println("]");
        
        // Test case 4: n = 5
        System.out.print("Test 4: [");
        for (int i = 0; i < 5; i++) {
            if (i > 0) System.out.print(",");
            System.out.print(rand10());
        }
        System.out.println("]");
        
        // Test case 5: n = 10
        System.out.print("Test 5: [");
        for (int i = 0; i < 10; i++) {
            if (i > 0) System.out.print(",");
            System.out.print(rand10());
        }
        System.out.println("]");
    }
}

File: dataset/ourMethod/CLAUDE_claude-sonnet-4-5-20250929/Task198.py

import random

def rand7() -> int:
    """Simulated rand7() for testing purposes"""
    return random.randint(1, 7)

def rand10() -> int:
    """Generate uniform random integer in range [1, 10] using rand7()"""
    while True:
        row = rand7()
        col = rand7()
        idx = (row - 1) * 7 + col
        if idx <= 40:
            return ((idx - 1) % 10) + 1

def main() -> None:
    # Test case 1: n = 1
    result = [rand10() for _ in range(1)]
    print(f"Test 1: {result}")
    
    # Test case 2: n = 2
    result = [rand10() for _ in range(2)]
    print(f"Test 2: {result}")
    
    # Test case 3: n = 3
    result = [rand10() for _ in range(3)]
    print(f"Test 3: {result}")
    
    # Test case 4: n = 5
    result = [rand10() for _ in range(5)]
    print(f"Test 4: {result}")
    
    # Test case 5: n = 10
    result = [rand10() for _ in range(10)]
    print(f"Test 5: {result}")

if __name__ == "__main__":
    main()
